### ISSUE TITLE
[Feature Request] OAuth2 Public Client Support

### ISSUE THREAD

**Issue by terorie** [Axosoft/glo-sdk#54]
TLDR: Is adding OAuth2 support for public clients (as opposed to closed servers) planned?

Hey, I know this is probably the wrong place to post, but I have a __feature request__ for the new Glo API.

As of now, the Glo OAuth2 API only supports the `Web Application Flow`, which assumes a trusted server run by the app developer to do all the API calls.

Client-side (e.g. by Android app) API calls aren't possible without revealing the API secret to users (which is bad practice) or asking the user to create a personal access token (bad UX).

OAuth2 provides two solutions:
 - [Implicit Flow](https://oauth.net/2/grant-types/implicit/)
 - Authorization Flow [PKCE extension](https://www.oauth.com/oauth2-servers/pkce/)

(Asking because I'm writing a CLI/CUI version of Glo, that calls the API directly)

**Comment by James-Quigley** [comment]
I don't think this is something we plan on adding. @jdgarcia might know more. 

I would recommend using the Web App Flow. It is possible to stand up a private server for free using something like Now.sh, Heroku, WebTasks etc.

As you mentioned, we also have support for [PATs](https://support.gitkraken.com/developers/pats/), so you could just prompt the user to provide your app a PAT with the correct scopes for a CLI app.
